lib/matplotlib/contour.py

```diff
@@ -921,6 +922,20 @@ def __init__(self, ax, *args, **kwargs):
             if self.allkinds is None:
                 self.allkinds = [None] * len(self.allsegs)
 
+            # get the edgecolors argument...
+            if self.edgecolors is None:
+                self.edgecolors = 'face'
+            if (not isinstance(self.edgecolors, six.string_types)
+                    or self.edgecolors not in ['face', 'none']):
+                raise ValueError(
+                        "contourf kwarg edgecolors must be either "
+                        "'face' or 'none'")
+            if self.edgecolors is 'face':
+                self.linewidths = 0.001
```

Comment:
Please use either `np.finfo(float).eps` or `np.nextafter(0, 1)` (otherwise we end up with a bunch of different definitions of "tiny number" throughout the codebase).

Reply:
Oh, hmmmm...  Well..  If I set line widths to smaller than 0.2 then I get a pdf that looks as above, but a png that looks like:

![testcontour](https://user-images.githubusercontent.com/1562854/32035316-32f92750-b9cd-11e7-9222-5e00ecee91ec.png)

Reply:
I don't get why that happens...